hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/Route.java

```diff
@@ -162,17 +177,52 @@ public void configure() {
 	            .when(simple("${in.header.messageType} == {{hibc-r50-endpoint}}"))
 	                .log("the HIBC endpoint (${in.header.messageType}) is reached and message will be dispatched to message queue(ENROL).")
                     .setBody(simple(SampleMessages.r50ResponseMessage))
-	            
+                 
 	            // others sending to JMB
 	            .otherwise()
                     .log("the JMB endpoint is reached and message will be dispatched to JMB!!")
-                    .setBody(simple(SampleMessages.r03ResponseMessage))
-            .end();
-           
+                	.process(new EventTimeProcessor())
+                	.wireTap("direct:messageSent")
+                		.endChoice()
+                    .setBody(simple(SampleMessages.R09_RESPONSE_MESSAGE))
+		            .log("Received response from JMB")
+		            .process(new EventTimeProcessor())
+		            .wireTap("direct:messageReceived")
+		            	.endChoice()
+            .end();        
+        
+        from("direct:start").log("wireTap route")
+        	.choice()
+				.when(header("isFileDropsEnabled").isEqualToIgnoreCase(Boolean.TRUE))
+				.bean(RequestFileDropGenerater.class).id("V2FileDropsRequest").log("wire tap done")
+			.end()
+			.choice()
+				.when(header("isAuditsEnabled").isEqualToIgnoreCase(Boolean.TRUE.toString()))
+					.process(new TransactionProcessor())
+					.setHeader(Util.HEADER_TRANSACTION_EVENT_TYPE).simple(TransactionEventType.TRANSACTION_START.name())					
+					.process(new TransactionEventProcessor())
+			.end();
         
-        from("direct:start").log("wireTap route").choice()
-			.when(header("isFileDropsEnabled").isEqualToIgnoreCase(Boolean.TRUE))
-			.bean(RequestFileDropGenerater.class).id("V2FileDropsRequest").log("wire tap done");
+		from("direct:transactionComplete").log("wireTap transactionComplete")
+			.choice()
+				.when(header("isAuditsEnabled").isEqualToIgnoreCase(Boolean.TRUE.toString()))
+					.setHeader(Util.HEADER_TRANSACTION_EVENT_TYPE).simple(TransactionEventType.TRANSACTION_COMPLETE.name())
+					.process(new TransactionEventProcessor())				
+			.end();
```

Comment:
Another option here might be to use a bean, instead of a processor, with multiple methods (one for each audit type) and just call the specific method you want for a transaction event. 
It's still a bunch of conditional logic but handled in the bean instead of the route which keeps the route a little cleaner



Saves you from needing to save the variables on the exchange as header parameters.